src/main/main.ts

```diff
@@ -173,11 +173,12 @@ electron.ipcMain.on(
       : urlString
 
     const openArguments: string[] = [
-      processedUrlTemplate,
       '-b',
       appId,
-      isAlt ? '--background' : '',
-    ].filter(Boolean)
+      ...(isAlt ? ['--background'] : []),
+      ...(isShift && app.privateArg ? ['--new', '--args', app.privateArg] : []),
+      processedUrlTemplate,
```

Comment:
Out of interest, does `processedUrlTemplate` need to be moved to the end?

Also, maybe we could use the new `.flat` method to keep syntax noise down?

```ts
const openArguments: string[] = [
  processedUrlTemplate,
  '-b',
  appId,
  isAlt ? '--background' : [],
  isShift && app.privateArg ? ['--new', '--args', app.privateArg] : []
].flat()
```

Reply:
Ah, yeah. I wanted to note this somewhere: because at least both Chrome and Firefox expect the URL as the parameter after `--private-window` or `--incognito`, like `--private-window <url>`.

By just moving it to the end of the argument list it works for the normal openings just like it did before, and if we're going with private mode then it will be appended to the very last argument for private/incognito mode.

I'll add a test to make sure that when selecting private/incognito mode we always get the URL passed as the last parameter.

Reply:
Ah, good to know!

Reply:
Actually, not sure how to write a test checking for that specifically, so I added a comment right above the line. Let me know if you have another idea!

Reply:
Yeah, that's enough for now 😄